src/Drivers/nlpPriDec_ex9_sparse_raja.cpp

```diff
@@ -0,0 +1,222 @@
+PriDecMasterProblemEx9Sparse::solve_master(hiopVector& x,
+                                           const bool& include_r,
+                                           const double& rval/*=0*/, 
+                                           const double* grad/*=0*/,
+                                           const double*hess /*=0*/,
+                                           const char* master_options_file/*=nullptr*/)
+{
+  obj_=-1e+20;
+  hiopSolveStatus status;
+  if(basecase_==NULL) {
+    basecase_ = new PriDecBasecaseProblemEx9(nx_);
+  }
+  basecase_->set_include(include_r);
+
+  if(include_r) {
+    assert(basecase_->quad_is_defined());
+  }
+  
+  hiopNlpSparse nlp(*basecase_, master_options_file);
+
+  //
+  // any of the options below can be overwritten by specifying them in the 'hiop_pridec_master.options' file
+  //
+  //nlp.options->SetStringValue("compute_mode", "hybrid");
+  //nlp.options->SetStringValue("dualsUpdateType", "linear");
+
+  //nlp.options->SetStringValue("fixed_var", "relax");
+  nlp.options->SetStringValue("Hessian", "analytical_exact");
+  nlp.options->SetStringValue("KKTLinsys", "xdycyd");
+  nlp.options->SetStringValue("compute_mode", "cpu");
+  //nlp.options->SetStringValue("compute_mode", "hybrid");
+  //nlp.options->SetStringValue("mem_space", mem_space.c_str());
+
+  nlp.options->SetIntegerValue("verbosity_level", 1);
+  nlp.options->SetNumericValue("mu0", 1e-1);
+  //nlp.options->SetNumericValue("tolerance", 1e-5);
+  
+  hiopAlgFilterIPMNewton solver(&nlp);
+
+  status = solver.run();
+
+  obj_ = solver.getObjective();
+  double* x_vec = x.local_data();
+  solver.getSolution(x_vec);
+
+  if(status<0) {
+    printf("solver returned negative solve status: %d (with objective is %18.12e)\n", status, solver.getObjective());
+    return status;
+  }
+  
+  // print x
+  // for(int i=0;i<nx_;i++) printf(" %d %18.12e ",i,x[i]);
+    
+  if(sol_==NULL) {
+    sol_ = new double[nx_];
+  }
+  memcpy(sol_,x_vec, nx_*sizeof(double));
+  
+  //compute the recourse estimate
+  if(include_r) {
+    double rec_appx = 0.;
+    basecase_->get_rec_obj(nx_, x_vec, rec_appx);
+    //printf("recourse estimate: is %18.12e\n", rec_appx); 
+  }
+  
+  return Solve_Success;
+  //return hiop::SolverInternal_Error;
+}
+
+bool PriDecMasterProblemEx9Sparse::
+set_recourse_approx_evaluator(const int n, 
+		              hiopInterfacePriDecProblem::RecourseApproxEvaluator* evaluator)
+{
+  assert(n==nc_);
+  basecase_->set_quadratic_terms(n, evaluator);
+  return true; 
+}
+
+// all the memory management is done through umpire in the PriDecRecourseProblemEx9Sparse class
+bool PriDecMasterProblemEx9Sparse::eval_f_rterm(size_t idx, const int& n, const double* x, double& rval)
+{
+  assert(nx_==n);
+  rval=-1e+20;
+  hiopSolveStatus status;
+  double* xi;
+  
+#ifdef HIOP_USE_MPI
+  double t3 =  MPI_Wtime(); 
+  double t4 = 0.; 
+#endif 
+  
+  // xi can be set below 
+  xi = new double[nS_]; 
+  for(int i=0;i<nS_;i++) xi[i] = 1.; 
```

Comment:
coding styleguide issues: need space inside the for, body of for inside brackets...

Reply:
Done.